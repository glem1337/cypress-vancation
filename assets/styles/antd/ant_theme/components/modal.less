// Modal
@modal-body-padding: 0;
@modal-header-bg: @component-background;
@modal-header-border-color-split: @border-color-split;
@modal-content-bg: @component-background;
@modal-heading-color: @heading-color;
@modal-footer-bg: transparent;
@modal-footer-border-color-split: @border-color-split;
@modal-footer-padding-vertical: 24px;
@modal-footer-padding-horizontal: 24px;
@modal-mask-bg: fade(black, 50%);
