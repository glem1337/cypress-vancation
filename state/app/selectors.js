import * as R from 'ramda';

export const prevPageSelector = R.path(['app', 'prevPage']);
export const currentPageSelector = R.path(['app', 'currentPage']);
export const currentCoordinatesSelector = R.path(['app', 'currentCoordinates']);
export const currentLocationSelector = R.path(['app', 'currentLocation']);
export const openGraphDataSelector = R.path(['app', 'openGraph']);
