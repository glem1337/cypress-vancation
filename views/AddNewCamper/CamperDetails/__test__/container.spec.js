import { shallow } from 'enzyme';
import * as R from 'ramda';

import diveTo from 'utils/testHelpers/diveToEnzyme';
import configureStore from 'redux-mock-store';
import describeValidationSchema from 'utils/testHelpers/describeValidationSchema';
import {
  fetchSpecificationsAction,
  fetchCamper,
  setEstimateEarningData,
} from 'state/concepts/camper/actions';
import CamperDetailsWrapper, { CamperDetailsContainer } from 'views/AddNewCamper/CamperDetails/container';
import {
  START_RANGE_LENGTH_CAMPER, FINISH_RANGE_LENGTH_CAMPER,
  START_RANGE_LENGTH_CAMPER_FLOAT, FINISH_RANGE_LENGTH_CAMPER_FLOAT, START_RANGE_AGE_CAMPER,
} from 'constants/camper';
import {
  mockedVehicleModel, mockedVehicleTypes, mockedVehicleMake,
} from 'views/__mocks__/mockSpecifications';
import { isUserLoggedInSelector } from 'state/concepts/session/selectors';
import mockedCamper from 'views/__mocks__/camper';

import { hideModal } from '../../../../state/modal/actions';

jest.mock('utils/form/isSubmitDisabled', () => jest.fn(() => true));

jest.mock('state/concepts/camper/selectors', () => ({
  vehicleMakeSelector: jest.fn(() => mockedVehicleMake),
  vehicleTypeSelector: jest.fn(() => mockedVehicleTypes),
  vehicleModelSelector: jest.fn(() => mockedVehicleModel),
  camperSelector: jest.fn(() => mockedCamper),
  camperCompletenessSelector: jest.fn(() => 22),
  isCamperExistSelector: jest.fn(() => true),
  isSidebarVisibleSelector: jest.fn(() => true),
  estimateEarningStateSelector: jest.fn(() => true),
}));

jest.mock('state/concepts/session/selectors', () => ({
  ...jest.requireActual('state/concepts/session/selectors'),
  isUserLoggedInSelector: jest.fn(() => true),
}));

jest.mock('state/data/selectors', () => ({
  loadingSelector: jest.fn(() => false),
}));

describe('CamperDetails Container', () => {
  let wrapper;
  let container;
  let instance;
  const store = configureStore()({});
  store.dispatch = jest.fn();
  store.logicMiddleware = {
    whenComplete: jest.fn(),
  };

  const defaultProps = {
    store,
    setValues: jest.fn(),
    setFieldValue: jest.fn(),
    camperId: 'test',
    router: {
      push: jest.fn(),
    },
    handleSubmit: jest.fn(),
  };

  beforeEach(() => {
    wrapper = shallow(<CamperDetailsWrapper {...defaultProps} />);
    container = diveTo(wrapper, CamperDetailsContainer);
    instance = container.instance();

    jest.clearAllMocks();
  });

  it('container snapshot', () => {
    expect(container).toMatchSnapshot();
  });

  it('has validation schema', () => {
    expect(describeValidationSchema(wrapper)).toMatchSnapshot();
  });

  describe('tests "getInitialProps" static method', () => {
    const ctx = {
      store,
      query: {
        camper_id: 'id',
      },
    };

    it('should fetch specifications', async () => {
      await CamperDetailsContainer.getInitialProps(ctx);

      expect(store.dispatch).toHaveBeenCalledTimes(1);
      expect(store.dispatch).toHaveBeenNthCalledWith(1, fetchSpecificationsAction());
    });

    it('should not fetch specifications', async () => {
      isUserLoggedInSelector.mockReturnValueOnce(false);

      await CamperDetailsContainer.getInitialProps(ctx);

      expect(store.dispatch).not.toHaveBeenCalled();
    });

    it('should fetch camper', async () => {
      const newCtx = {
        store,
        query: {
          camper_id: 'test id',
        },
      };

      await CamperDetailsContainer.getInitialProps(newCtx);

      expect(store.dispatch).toHaveBeenCalledWith(fetchCamper('test id', 'specification_detail'));
      expect(store.logicMiddleware.whenComplete).toHaveBeenCalled();
    });
  });

  it('tests "isLoading" instance getter', () => {
    expect(instance.isLoading).toEqual(false);
  });

  it('tests "submitData" instance getter', () => {
    const { camperId } = container.props();

    const result = R.compose(
      R.append({ camperId }),
      R.props(['values', 'setErrors', 'setSubmitting', 'resetForm', 'setErrors', 'setValues']),
    )(container.props());

    expect(instance.submitData).toEqual(result);
  });

  it('tests "getYears" instance method', () => {
    const exp = R.range(START_RANGE_AGE_CAMPER, new Date().getFullYear() + 1)
      .map(elem => ({ value: `${elem}`, label: `${elem}` }))
      .reverse();

    const result = instance.getYears();

    expect(result).toEqual(exp);
  });

  it('tests "getCamperLength" instance method', () => {
    const exp = [];
    const floatLength = R.range(START_RANGE_LENGTH_CAMPER_FLOAT, FINISH_RANGE_LENGTH_CAMPER_FLOAT);

    for (let start = START_RANGE_LENGTH_CAMPER; start <= FINISH_RANGE_LENGTH_CAMPER; start += 1) {
      exp.push({ value: `${start}`, label: `${start} \`` });
      if (floatLength.includes(start)) {
        exp.push({ value: `${start + 0.5}`, label: `${start + 0.5} \`` });
      }
    }

    const result = instance.getCamperLength();

    expect(result).toEqual(exp);
  });

  it('tests "currentMake" instance method', () => {
    const currentModelId = mockedVehicleModel[0].id;
    const models = ({ vehicleModels, id }) => ({
      id,
      models: R.map(elem => R.prop(['id'], elem), vehicleModels),
    });
    const getMake = elem => elem.models.some(model => model === currentModelId);
    const { id } = R.find(getMake, R.map(models, mockedVehicleMake));

    expect(instance.currentMake(currentModelId)).toEqual({ id, name: mockedVehicleMake[0].name });
  });

  it('tests "currentType" instance method', () => {
    const currentMakeId = mockedVehicleMake[0].id;
    const currentModel = R.compose(
      R.find(({ makes }) => makes.some(make => make === currentMakeId)),
      R.map(({ vehicleMakes, id }) => ({ id, makes: R.map(elem => R.prop(['id'], elem), vehicleMakes) })),
    )(mockedVehicleTypes);

    expect(instance.currentType(currentMakeId))
      .toEqual({ id: currentModel.id, name: mockedVehicleTypes[0].name });
  });

  it('tests "firstModel" instance method', () => {
    const currentMakeName = mockedVehicleMake[0].name;
    const value = R.compose(
      R.prop(0),
      R.propOr([], 'vehicleModels'),
      R.find(item => item.name === currentMakeName),
      R.defaultTo([]),
    )(mockedVehicleMake);

    expect(instance.firstModel(currentMakeName)).toEqual(value);
  });

  it('tests "handlerMake" instance method', () => {
    const currentItem = { value: mockedVehicleMake[0].name, name: 'name' };
    const setItem = ({ id, name, iconUrl }) => ({ id, value: name, img: iconUrl, label: name });

    const currentTypeSpyOn = jest.spyOn(instance, 'currentType');
    const firstModelSpyOn = jest.spyOn(instance, 'firstModel');

    const prepareMakeItem = R.map(setItem, mockedVehicleMake);
    const currentMake = R.find(modelElem => modelElem.label === currentItem.value, prepareMakeItem);

    instance.handlerMake(currentItem, prepareMakeItem);

    expect(currentTypeSpyOn).toHaveBeenCalledTimes(1);
    expect(currentTypeSpyOn).toHaveBeenCalledWith(currentMake.id);

    expect(firstModelSpyOn).toHaveBeenCalledTimes(1);
    expect(firstModelSpyOn).toHaveBeenCalledWith(currentItem.value);
  });

  it('tests "vehicleTypes" instance getter', () => {
    const exp = R.compose(
      R.map(({
        id, name, iconUrl, estimatedEarning,
      }) => ({
        id, value: name, img: iconUrl, label: name, estimatedEarning,
      })),
    )(mockedVehicleTypes);

    expect(instance.vehicleTypes).toEqual(exp);
  });

  it('tests "vehicleMake" instance getter', () => {
    const prepareMakes = R.compose(
      R.map(({ id, name, iconUrl }) => ({ id, value: name, img: iconUrl, label: name })),
    )(mockedVehicleMake);

    const exp = R.unionWith(R.eqBy(R.prop('value')), prepareMakes, prepareMakes);

    expect(instance.vehicleMake).toEqual(exp);
  });

  it('tests "builtCamper" instance getter', () => {
    const exp = mockedVehicleModel
      .map(({ builderName }) => ({ value: builderName, label: builderName }));

    expect(instance.builtCamper).toEqual(exp);
  });

  describe('tests "vehicleModel" instance getter', () => {
    it('without filter', () => {
      const model = R.compose(
        R.sortBy(R.prop('value')),
        R.map(({
           id, name, builderName, insideHeigh, length,
        }) => ({
            id, value: name, label: name, builderName, insideHeigh, length,
        })),
      )(mockedVehicleModel);

      expect(instance.vehicleModel).toEqual(model);
    });

    it('with filter', () => {
      const nameMake = mockedVehicleMake[0].name;

      container.setProps({ values: { ...defaultProps.values, name: nameMake } });

      const filteredModels = R.compose(
        R.map(elem => elem.id),
        R.propOr([], 'vehicleModels'),
        R.find(elem => elem.name === nameMake),
      )(mockedVehicleMake);

      const model = R.compose(
        R.sortBy(R.prop('value')),
        R.filter(elem => filteredModels.includes(elem.id)),
        R.map(({
          id, name, builderName, insideHeigh, length,
        }) => ({
          id, value: name, label: name, builderName, insideHeigh, length,
        })),
      )(mockedVehicleModel);

      expect(instance.vehicleModel).toEqual(model);
    });
  });

  it('tests "componentDidUpdate" instance method', () => {
    const estimateEarning = {
      name: mockedVehicleTypes[0].name,
      iconUrl: mockedVehicleTypes[0].iconUrl,
      estimatedEarning: mockedVehicleTypes[0].estimatedEarning,
    };

    container.setProps({ values: {
        vehicleTypeName: mockedVehicleTypes[0].name,
      } });

    instance.componentDidUpdate({
      values: {
        vehicleTypeName: '',
      },
    });

    expect(store.dispatch).toHaveBeenCalledWith(setEstimateEarningData(estimateEarning));
  });

  describe('tests `isFormValid` instance getter', () => {
    it('when camper id is correct', () => {
      container.setProps({ camperId: 'test' });

      const { isFormValid } = instance;

      expect(isFormValid).toBe(true);
    });

    it('when camper id is incorrect', () => {
      container.setProps({ camperId: 'id' });

      const { isFormValid } = instance;

      expect(isFormValid).toBe(false);
    });
  });

  describe('tests `leavePage` instance method', () => {
    it('with saving', async () => {
      await instance.leavePage(true, 'test')();

      expect(store.dispatch).toHaveBeenCalledWith(hideModal());
    });

    it('without saving', async () => {
      await instance.leavePage(false, 'test')();

      expect(defaultProps.router.push).toHaveBeenCalledWith('test');

      expect(store.dispatch).toHaveBeenCalledWith(hideModal());
    });
  });
});
