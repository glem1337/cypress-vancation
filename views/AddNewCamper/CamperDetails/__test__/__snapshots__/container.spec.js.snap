// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CamperDetails Container container snapshot 1`] = `
<CamperDetails
  builtCamper={
    Array [
      Object {
        "label": "Santina McGlynn",
        "value": "Santina McGlynn",
      },
      Object {
        "label": "Catheryn Haley",
        "value": "Catheryn Haley",
      },
    ]
  }
  camper={
    Object {
      "blockedPeriods": Array [],
      "camperAddition": Object {
        "amenityHealthSafetyItems": Array [
          Object {
            "healthSafety": Object {
              "iconUrl": "path/to/icon/url",
              "id": "74a9c587-7b78-4aa5-ba93-ba3d310be127",
              "name": "name",
            },
          },
        ],
        "camperQuestions": Array [
          Object {
            "id": "74a9c587-7b78-4aa5-ba93-ba3d310be127",
            "required": true,
            "text": "test",
          },
        ],
        "customRestrictionRoads": Array [
          Object {
            "active": true,
            "id": "74a9c587-7b78-4aa5-ba93-ba3d310be127",
            "name": "name",
          },
        ],
        "customRestrictionRules": Array [
          Object {
            "active": true,
            "id": "74a9c587-7b78-4aa5-ba93-ba3d310be127",
            "name": "name",
          },
        ],
        "customTravelAccessories": Array [
          Object {
            "active": true,
            "description": "",
            "id": "dd477886-ec22-4da9-ba4c-08627f35fb1a",
            "maxAmount": "2.0",
            "name": "Bike Rack",
            "price": "1.0",
            "priceUnit": "each",
          },
        ],
        "customTravelRestrictions": Array [
          Object {
            "active": true,
            "id": "74a9c587-7b78-4aa5-ba93-ba3d310be127",
            "name": "name",
          },
        ],
        "restrictionRoad": Object {
          "dirtryRoad": true,
          "fourWheelRoad": false,
          "offRoad": false,
          "snowAndIceRoad": false,
        },
        "restrictionRule": Object {
          "allowPets": false,
          "festivalApproved": false,
          "smoking": false,
        },
        "travelRestriction": Object {
          "burningMan": false,
          "canada": true,
          "mexico": false,
        },
      },
      "camperCalendar": Object {
        "blockedPeriods": Array [
          Object {
            "endDate": "2021-08-06",
            "id": "56ceb535-322f-42c6-8dcd-a88ad59000f3",
            "startDate": "2021-08-05",
          },
          Object {
            "endDate": "2021-08-25",
            "id": "17a32ec0-a486-4247-910c-6c0c57ba6ec3",
            "startDate": "2021-08-25",
          },
          Object {
            "endDate": "2021-07-28",
            "id": "2c7bbaef-6b00-494f-b005-6a762e90c785",
            "startDate": "2021-07-28",
          },
        ],
        "customDiscountPeriods": Array [
          Object {
            "endDate": "2021-07-28",
            "id": "ffb3f0cc-4513-4fdc-81d0-03587232d9ee",
            "monthlyDiscountPercent": 23,
            "startDate": "2021-07-27",
            "weeklyDiscountPercent": 15,
          },
          Object {
            "endDate": "2021-07-28",
            "id": "a070339d-7058-4769-8e16-814084c6bfe3",
            "monthlyDiscountPercent": 23,
            "startDate": "2021-07-28",
            "weeklyDiscountPercent": 15,
          },
          Object {
            "endDate": "2021-07-28",
            "id": "aba16cba-5873-4273-8172-816b276a2db7",
            "monthlyDiscountPercent": 44,
            "startDate": "2021-07-28",
            "weeklyDiscountPercent": 44,
          },
          Object {
            "endDate": "2021-07-29",
            "id": "586016b9-d351-4ca1-ac8a-373e7e3b8832",
            "monthlyDiscountPercent": 33,
            "startDate": "2021-07-29",
            "weeklyDiscountPercent": 15,
          },
        ],
        "externalCalendars": Array [],
        "id": "e1db81b1-8ad5-4e88-9c99-f5c9020148eb",
        "pricingPeriods": Array [
          Object {
            "endDate": "2021-07-28",
            "id": "8aa92a51-ef00-4dc6-8b84-2820869b9278",
            "price": "33.0",
            "startDate": "2021-07-28",
          },
          Object {
            "endDate": "2021-07-30",
            "id": "248ba38e-e18a-467c-996b-36aa4bd73e85",
            "price": "320.0",
            "startDate": "2021-07-29",
          },
        ],
      },
      "camperPhotos": Array [
        Object {
          "id": "74a9c587-7b78-4aa5-ba93-ba3d310be127",
          "photoUrl1100": "photoUrl1100",
        },
      ],
      "camperRule": Object {
        "bookingApprovalPolicy": "instant_book",
        "cancellationPolicy": "easy_going",
        "id": "74a9c587-7b78-4aa5-ba93-ba3d310be127",
      },
      "deliveryInformation": Object {
        "id": "74a9c587-7b78-4aa5-ba93-ba3d310be127",
        "pickup": true,
      },
      "description": "desc",
      "estimatedEarning": 1200,
      "externalCalendars": Array [],
      "glamper": true,
      "id": "a18f7338-c2b3-4cee-be4d-579c0a726c1a",
      "insurance": "pending",
      "insuranceInfo": Object {
        "id": "74a9c587-7b78-4aa5-ba93-ba3d310be127",
      },
      "latitude": 40.772,
      "longitude": -73.93030000000002,
      "name": "test",
      "owner": Object {
        "account": Object {
          "createdAt": "2021-07-09T13:39:26.587Z",
          "drivingLicenceVerified": false,
          "email": "kateryna.sessa+owner@rubygarage.org",
          "emailVerified": false,
          "id": "12563773-6e8f-4861-8abd-82a78632c6ac",
          "user": Object {
            "avatarUrl": null,
            "createdAt": "2021-07-09T13:39:26.590Z",
            "firstName": "Kat",
            "id": "77015afb-71e3-4b94-a9cb-69a97dc6b358",
            "lastName": "Kat",
            "updatedAt": "2021-07-09T13:39:26.590Z",
          },
        },
        "avatarUrl": null,
        "averageRating": null,
        "businessTitle": null,
        "campersCount": 25,
        "createdAt": "Member since July 2021",
        "description": "description",
        "email": null,
        "id": "74a9c587-7b78-4aa5-ba93-ba3d310be127",
        "idVerified": false,
        "type": "owner",
      },
      "place": "Astoria, New York, New York 11102, United States",
      "placeId": "neighborhood.290416",
      "pricingInfo": Object {
        "id": "74a9c587-7b78-4aa5-ba93-ba3d310be127",
      },
      "pricingPeriods": Array [],
      "publicId": "111",
      "specificationDetail": Object {
        "drivetrain": "4x4",
        "fuelType": "electric",
        "insideHeight": "low_top",
        "length": "67.89",
        "mileage": "Under 50,000",
        "modelNaming": "Clair Weimann",
        "name": "Katelyn Lowe",
        "seats": 1,
        "sleeps": 1,
        "stateRegistred": "Arkansas",
        "transmission": "automatice",
        "whoBuiltCamper": "Vasya",
        "year": 1585,
      },
      "status": "draft",
      "tripFee": Object {
        "cleaning": 1,
        "customFees": Array [
          Object {
            "frequency": "per_day",
            "id": "id",
            "name": "name",
            "price": 5,
          },
        ],
        "dumpingFee": 2,
        "flue": 3,
        "id": "74a9c587-7b78-4aa5-ba93-ba3d310be127",
        "lateDropOff": 4,
        "tripFeeGenerator": Object {
          "available": 8,
          "limit": true,
          "overage": 9,
        },
        "tripFeeMileage": Object {
          "available": 6,
          "limit": true,
          "overage": 7,
        },
      },
      "vehicleTypeIconUrl": "test.svg",
      "vehicleTypeName": "Modern Van",
    }
  }
  camperCompleteness={22}
  camperId="test"
  camperLength={
    Array [
      Object {
        "label": "10 \`",
        "value": "10",
      },
      Object {
        "label": "11 \`",
        "value": "11",
      },
      Object {
        "label": "12 \`",
        "value": "12",
      },
      Object {
        "label": "13 \`",
        "value": "13",
      },
      Object {
        "label": "14 \`",
        "value": "14",
      },
      Object {
        "label": "15 \`",
        "value": "15",
      },
      Object {
        "label": "16 \`",
        "value": "16",
      },
      Object {
        "label": "16.5 \`",
        "value": "16.5",
      },
      Object {
        "label": "17 \`",
        "value": "17",
      },
      Object {
        "label": "17.5 \`",
        "value": "17.5",
      },
      Object {
        "label": "18 \`",
        "value": "18",
      },
      Object {
        "label": "18.5 \`",
        "value": "18.5",
      },
      Object {
        "label": "19 \`",
        "value": "19",
      },
      Object {
        "label": "19.5 \`",
        "value": "19.5",
      },
      Object {
        "label": "20 \`",
        "value": "20",
      },
      Object {
        "label": "20.5 \`",
        "value": "20.5",
      },
      Object {
        "label": "21 \`",
        "value": "21",
      },
      Object {
        "label": "21.5 \`",
        "value": "21.5",
      },
      Object {
        "label": "22 \`",
        "value": "22",
      },
      Object {
        "label": "22.5 \`",
        "value": "22.5",
      },
      Object {
        "label": "23 \`",
        "value": "23",
      },
      Object {
        "label": "23.5 \`",
        "value": "23.5",
      },
      Object {
        "label": "24 \`",
        "value": "24",
      },
      Object {
        "label": "24.5 \`",
        "value": "24.5",
      },
      Object {
        "label": "25 \`",
        "value": "25",
      },
      Object {
        "label": "25.5 \`",
        "value": "25.5",
      },
      Object {
        "label": "26 \`",
        "value": "26",
      },
      Object {
        "label": "26.5 \`",
        "value": "26.5",
      },
      Object {
        "label": "27 \`",
        "value": "27",
      },
      Object {
        "label": "27.5 \`",
        "value": "27.5",
      },
      Object {
        "label": "28 \`",
        "value": "28",
      },
      Object {
        "label": "28.5 \`",
        "value": "28.5",
      },
      Object {
        "label": "29 \`",
        "value": "29",
      },
      Object {
        "label": "29.5 \`",
        "value": "29.5",
      },
      Object {
        "label": "30 \`",
        "value": "30",
      },
      Object {
        "label": "31 \`",
        "value": "31",
      },
      Object {
        "label": "32 \`",
        "value": "32",
      },
      Object {
        "label": "33 \`",
        "value": "33",
      },
      Object {
        "label": "34 \`",
        "value": "34",
      },
      Object {
        "label": "35 \`",
        "value": "35",
      },
      Object {
        "label": "36 \`",
        "value": "36",
      },
      Object {
        "label": "37 \`",
        "value": "37",
      },
      Object {
        "label": "38 \`",
        "value": "38",
      },
      Object {
        "label": "39 \`",
        "value": "39",
      },
      Object {
        "label": "40 \`",
        "value": "40",
      },
    ]
  }
  dirty={false}
  errors={Object {}}
  estimateEarningState={true}
  getFieldHelpers={[Function]}
  getFieldMeta={[Function]}
  getFieldProps={[Function]}
  handleBlur={[Function]}
  handleChange={[Function]}
  handleReset={[Function]}
  handleSubmit={[Function]}
  handlerMake={[Function]}
  hideModal={[Function]}
  initialErrors={Object {}}
  initialTouched={Object {}}
  initialValues={
    Object {
      "camperLocation": Object {
        "id": "neighborhood.290416",
        "latitude": 40.772,
        "longitude": -73.93030000000002,
        "place": "Astoria, New York, New York 11102, United States",
      },
      "drivetrain": "4x4",
      "fuelType": "electric",
      "insideHeight": "low_top",
      "length": "67",
      "mileage": "Under 50,000",
      "modelNaming": "Clair Weimann",
      "name": "Katelyn Lowe",
      "seats": "1",
      "sleeps": "1",
      "stateRegistred": "Arkansas",
      "transmission": "automatice",
      "vehicleTypeName": "Modern Van",
      "whoBuiltCamper": "Vasya",
      "year": "1585",
    }
  }
  intl={
    Object {
      "formatMessage": [Function],
    }
  }
  isFormValid={true}
  isLoadingCreate={false}
  isLoadingUpdate={false}
  isSubmitting={false}
  isValid={true}
  isValidating={false}
  onSubmit={[Function]}
  registerField={[Function]}
  resetForm={[Function]}
  router={
    Object {
      "push": [MockFunction],
    }
  }
  setErrors={[Function]}
  setEstimate={[Function]}
  setFieldError={[Function]}
  setFieldTouched={[Function]}
  setFieldValue={[Function]}
  setFormikState={[Function]}
  setLeavePageMethod={[Function]}
  setStatus={[Function]}
  setSubmitting={[Function]}
  setTouched={[Function]}
  setValues={[Function]}
  showModal={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [MockFunction],
      "getActions": [Function],
      "getState": [Function],
      "logicMiddleware": Object {
        "whenComplete": [MockFunction],
      },
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  submitCount={0}
  submitForm={[Function]}
  touched={Object {}}
  unregisterField={[Function]}
  validateField={[Function]}
  validateForm={[Function]}
  validateOnBlur={true}
  validateOnChange={true}
  validateOnMount={false}
  values={
    Object {
      "camperLocation": Object {
        "id": "neighborhood.290416",
        "latitude": 40.772,
        "longitude": -73.93030000000002,
        "place": "Astoria, New York, New York 11102, United States",
      },
      "drivetrain": "4x4",
      "fuelType": "electric",
      "insideHeight": "low_top",
      "length": "67",
      "mileage": "Under 50,000",
      "modelNaming": "Clair Weimann",
      "name": "Katelyn Lowe",
      "seats": "1",
      "sleeps": "1",
      "stateRegistred": "Arkansas",
      "transmission": "automatice",
      "vehicleTypeName": "Modern Van",
      "whoBuiltCamper": "Vasya",
      "year": "1585",
    }
  }
  vehicleMake={
    Array [
      Object {
        "id": "cf045637-c752-4690-8a82-89f4f0c5d69c",
        "label": "Jonie Altenwerth",
        "value": "Jonie Altenwerth",
      },
    ]
  }
  vehicleModel={
    Array [
      Object {
        "builderName": "Catheryn Haley",
        "id": "fc7b3921-d1fb-4903-8830-9018167e5ba6",
        "insideHeigh": "Laraine Breitenberg",
        "label": "Karoline Pollich",
        "length": "19.0",
        "value": "Karoline Pollich",
      },
      Object {
        "builderName": "Santina McGlynn",
        "id": "d9d1f129-c11f-4893-b887-c61434c9940b",
        "insideHeigh": "Dodie Stracke",
        "label": "Thao Mitchell",
        "length": "86.9",
        "value": "Thao Mitchell",
      },
    ]
  }
  vehicleType={
    Array [
      Object {
        "estimatedEarning": 1200,
        "iconUrl": ".svg",
        "id": "5090dd03-b9f7-461d-8351-f120b71dd07c",
        "name": "Kiley Cruickshank",
        "vehicleMakes": Array [
          Object {
            "id": "cf045637-c752-4690-8a82-89f4f0c5d69c",
            "name": "Jonie Altenwerth",
            "vehicleModels": Array [
              Object {
                "builderName": "Santina McGlynn",
                "id": "d9d1f129-c11f-4893-b887-c61434c9940b",
                "insideHeigh": "Dodie Stracke",
                "length": "86.9",
                "name": "Thao Mitchell",
                "year": 1281,
              },
              Object {
                "builderName": "Catheryn Haley",
                "id": "fc7b3921-d1fb-4903-8830-9018167e5ba6",
                "insideHeigh": "Laraine Breitenberg",
                "length": "19.0",
                "name": "Karoline Pollich",
                "year": 882,
              },
            ],
          },
        ],
      },
    ]
  }
  vehicleTypes={
    Array [
      Object {
        "estimatedEarning": 1200,
        "id": "5090dd03-b9f7-461d-8351-f120b71dd07c",
        "img": ".svg",
        "label": "Kiley Cruickshank",
        "value": "Kiley Cruickshank",
      },
    ]
  }
  years={
    Array [
      Object {
        "label": "2000",
        "value": "2000",
      },
      Object {
        "label": "1999",
        "value": "1999",
      },
      Object {
        "label": "1998",
        "value": "1998",
      },
      Object {
        "label": "1997",
        "value": "1997",
      },
      Object {
        "label": "1996",
        "value": "1996",
      },
      Object {
        "label": "1995",
        "value": "1995",
      },
      Object {
        "label": "1994",
        "value": "1994",
      },
      Object {
        "label": "1993",
        "value": "1993",
      },
      Object {
        "label": "1992",
        "value": "1992",
      },
      Object {
        "label": "1991",
        "value": "1991",
      },
      Object {
        "label": "1990",
        "value": "1990",
      },
      Object {
        "label": "1989",
        "value": "1989",
      },
      Object {
        "label": "1988",
        "value": "1988",
      },
      Object {
        "label": "1987",
        "value": "1987",
      },
      Object {
        "label": "1986",
        "value": "1986",
      },
      Object {
        "label": "1985",
        "value": "1985",
      },
      Object {
        "label": "1984",
        "value": "1984",
      },
      Object {
        "label": "1983",
        "value": "1983",
      },
      Object {
        "label": "1982",
        "value": "1982",
      },
      Object {
        "label": "1981",
        "value": "1981",
      },
      Object {
        "label": "1980",
        "value": "1980",
      },
      Object {
        "label": "1979",
        "value": "1979",
      },
      Object {
        "label": "1978",
        "value": "1978",
      },
      Object {
        "label": "1977",
        "value": "1977",
      },
      Object {
        "label": "1976",
        "value": "1976",
      },
      Object {
        "label": "1975",
        "value": "1975",
      },
      Object {
        "label": "1974",
        "value": "1974",
      },
      Object {
        "label": "1973",
        "value": "1973",
      },
      Object {
        "label": "1972",
        "value": "1972",
      },
      Object {
        "label": "1971",
        "value": "1971",
      },
      Object {
        "label": "1970",
        "value": "1970",
      },
    ]
  }
/>
`;

exports[`CamperDetails Container has validation schema 1`] = `
Object {
  "fields": Object {
    "camperLocation": Object {
      "fields": Object {},
      "label": undefined,
      "meta": undefined,
      "notOneOf": Array [],
      "oneOf": Array [],
      "tests": Array [
        Object {
          "name": "geocoder",
          "params": undefined,
        },
      ],
      "type": "object",
    },
    "drivetrain": Object {
      "label": undefined,
      "meta": undefined,
      "notOneOf": Array [],
      "oneOf": Array [],
      "tests": Array [
        Object {
          "name": "required",
          "params": undefined,
        },
      ],
      "type": "string",
    },
    "fuelType": Object {
      "label": undefined,
      "meta": undefined,
      "notOneOf": Array [],
      "oneOf": Array [],
      "tests": Array [
        Object {
          "name": "required",
          "params": undefined,
        },
      ],
      "type": "string",
    },
    "insideHeight": Object {
      "label": undefined,
      "meta": undefined,
      "notOneOf": Array [],
      "oneOf": Array [],
      "tests": Array [
        Object {
          "name": "required",
          "params": undefined,
        },
      ],
      "type": "string",
    },
    "length": Object {
      "label": undefined,
      "meta": undefined,
      "notOneOf": Array [],
      "oneOf": Array [],
      "tests": Array [
        Object {
          "name": "required",
          "params": undefined,
        },
        Object {
          "name": "min",
          "params": Object {
            "min": 10,
          },
        },
        Object {
          "name": "max",
          "params": Object {
            "max": 40,
          },
        },
      ],
      "type": "number",
    },
    "mileage": Object {
      "label": undefined,
      "meta": undefined,
      "notOneOf": Array [],
      "oneOf": Array [],
      "tests": Array [
        Object {
          "name": "required",
          "params": undefined,
        },
      ],
      "type": "string",
    },
    "modelNaming": Object {
      "label": undefined,
      "meta": undefined,
      "notOneOf": Array [],
      "oneOf": Array [],
      "tests": Array [
        Object {
          "name": "required",
          "params": undefined,
        },
      ],
      "type": "string",
    },
    "name": Object {
      "label": undefined,
      "meta": undefined,
      "notOneOf": Array [],
      "oneOf": Array [],
      "tests": Array [
        Object {
          "name": "required",
          "params": undefined,
        },
      ],
      "type": "string",
    },
    "seats": Object {
      "label": undefined,
      "meta": undefined,
      "notOneOf": Array [],
      "oneOf": Array [],
      "tests": Array [
        Object {
          "name": "required",
          "params": undefined,
        },
        Object {
          "name": "min",
          "params": Object {
            "min": 1,
          },
        },
        Object {
          "name": "max",
          "params": Object {
            "max": 20,
          },
        },
      ],
      "type": "number",
    },
    "sleeps": Object {
      "label": undefined,
      "meta": undefined,
      "notOneOf": Array [],
      "oneOf": Array [],
      "tests": Array [
        Object {
          "name": "required",
          "params": undefined,
        },
        Object {
          "name": "min",
          "params": Object {
            "min": 1,
          },
        },
        Object {
          "name": "max",
          "params": Object {
            "max": 20,
          },
        },
      ],
      "type": "number",
    },
    "stateRegistred": Object {
      "label": undefined,
      "meta": undefined,
      "notOneOf": Array [],
      "oneOf": Array [],
      "tests": Array [
        Object {
          "name": "required",
          "params": undefined,
        },
      ],
      "type": "string",
    },
    "transmission": Object {
      "label": undefined,
      "meta": undefined,
      "notOneOf": Array [],
      "oneOf": Array [],
      "tests": Array [
        Object {
          "name": "required",
          "params": undefined,
        },
      ],
      "type": "string",
    },
    "vehicleTypeName": Object {
      "label": undefined,
      "meta": undefined,
      "notOneOf": Array [],
      "oneOf": Array [],
      "tests": Array [
        Object {
          "name": "required",
          "params": undefined,
        },
      ],
      "type": "string",
    },
    "whoBuiltCamper": Object {
      "label": undefined,
      "meta": undefined,
      "notOneOf": Array [],
      "oneOf": Array [],
      "tests": Array [],
      "type": "string",
    },
    "year": Object {
      "label": undefined,
      "meta": undefined,
      "notOneOf": Array [],
      "oneOf": Array [],
      "tests": Array [
        Object {
          "name": "required",
          "params": undefined,
        },
      ],
      "type": "string",
    },
  },
  "label": undefined,
  "meta": undefined,
  "notOneOf": Array [],
  "oneOf": Array [],
  "tests": Array [],
  "type": "object",
}
`;
